pkg/setup/setup.go

```diff
@@ -35,7 +35,12 @@ func SetupBacalhauRepo(repoDir string) (*repo.FsRepo, error) {
 		return nil, fmt.Errorf("failed to create repo: %w", err)
 	}
 	if exists, err := fsRepo.Exists(); err != nil {
-		return nil, fmt.Errorf("failed to check if repo exists: %w", err)
+		switch err.(type) {
+		case repo.ErrUnknownRepoVersion:
```

Comment:
Would it be worth following a similar pattern to how `os.IsNotExist` works? i.e. maybe something like: `repo.IsUnknownVersion`?

Reply:
Have implemented in 0e7f06cfe
Didn't think I was going to like it, but it saved on the type switch and so yeah, much better.